app/assets/javascripts/discourse/models/bookmark.js.es6

```diff
@@ -18,6 +22,88 @@ const Bookmark = RestModel.extend({
     return ajax(this.url, {
       type: "DELETE"
     });
+  },
+
+  @discourseComputed("highest_post_number", "url")
+  lastPostUrl(highestPostNumber) {
+    return this.urlForPostNumber(highestPostNumber);
+  },
+
+  // Helper to build a Url with a post number
+  urlForPostNumber(postNumber) {
+    let url = Discourse.getURL(`/t/${this.topic_id}`);
+    if (postNumber && postNumber > 0) {
+      url += `/${postNumber}`;
+    }
+    return url;
+  },
+
+  // returns createdAt if there's no bumped date
+  @discourseComputed("bumped_at", "createdAt")
+  bumpedAt(bumped_at, createdAt) {
+    if (bumped_at) {
+      return new Date(bumped_at);
+    } else {
+      return createdAt;
+    }
+  },
+
+  @discourseComputed("bumpedAt", "createdAt")
+  bumpedAtTitle(bumpedAt, createdAt) {
+    const firstPost = I18n.t("first_post");
+    const lastPost = I18n.t("last_post");
+    const createdAtDate = longDate(createdAt);
+    const bumpedAtDate = longDate(bumpedAt);
+
+    return `${firstPost}: ${createdAtDate}\n${lastPost}: ${bumpedAtDate}`;
+  },
+
+  @discourseComputed("created_at")
+  createdAt(created_at) {
+    return new Date(created_at);
+  },
+
+  @discourseComputed("tags")
+  visibleListTags(tags) {
+    if (!tags || !Discourse.SiteSettings.suppress_overlapping_tags_in_list) {
```

Comment:
Is `this.siteSettings` not available in your model? It's preferred to the global constant.